CHANGELOG.md

```diff
@@ -47,6 +47,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added required `array` argument when invoking `SerializationError` ([#419](https://github.com/opensearch-project/opensearch-js/pull/419))
 - Removed unnecessary `data` argument when invoking `OpenSearchClientError` ([#421](https://github.com/opensearch-project/opensearch-js/pull/421))
 - Fixed typos in `ConnectionPool` ([#427](https://github.com/opensearch-project/opensearch-js/pull/427))
+- Fixed typo in connection `_agent` ([#443](https://github.com/opensearch-project/opensearch-js/issues/443)
```

Comment:
You can skip CHANGELOGs for trivial things :)

Reply:
> I don't quite understand what the side effects of having `.agent` was. What was broken? Is there a test that you can write that shows broken behavior (that's fixed with this change)?

In the file [Connection.d.ts](https://github.com/opensearch-project/opensearch-js/blob/main/lib/Connection.d.ts#LL103C3-L103C87), the `agent` property is currently declared as `_agent`,  which has caused confusion.  It might be better to update the `Connection` interface to use `agent` instead, since it is not actually an internal property

@Naurabay 
```diff
export default class Connection {
  ...
  _status: string;
+ agent: http.Agent | https.Agent | hpagent.HttpProxyAgent | hpagent.HttpsProxyAgent;
- _agent: http.Agent | https.Agent | hpagent.HttpProxyAgent | hpagent.HttpsProxyAgent;
  constructor(opts?: ConnectionOptions) {
  ...
}
```